src/ngx_http_lua_bodyfilterby.c

```diff
@@ -140,6 +146,19 @@ ngx_http_lua_body_filter_by_chunk(lua_State *L, ngx_http_request_t *r,
 
     lua_settop(L, 0);
 
+    if (ctx->body_exited) {
+        /* send last content */
+        for (cl = in; cl->next; cl = cl->next) {
+            /* do nothing */
+        }
+        /* make ngx_http_write_filter_module happy */
+        cl->buf->last_buf = 1;
+
+        ctx->eof = 1;
+
+        return NGX_OK;
```

Comment:
@rainingmaster Also, according to our and nginx's convention, returning (or exiting with) `NGX_OK` usually means exiting the current *handler invocation* successfully, instead of exiting the whole request successfully. The only exception is the content phase handler, which is special anyway. In this context, returning (or exiting with) `NGX_OK` from a body filter handler should not end the whole response body, but just quit the current body filter handler invocation, but it should not prevent the current body filter handler from being invoked again in the same request.

BTW, I'm having problems reading your English text in your comment above. I'm just assuming that I understand your comment and intention correctly here, with reference to your patch here.